core/trino-main/src/main/java/io/trino/execution/RenameTableTask.java

```diff
@@ -103,6 +103,12 @@ public ListenableFuture<Void> execute(
         if (metadata.getTableHandle(session, target).isPresent()) {
             throw semanticException(TABLE_ALREADY_EXISTS, statement, "Target table '%s' already exists", target);
         }
+        if (metadata.isView(session, target)) {
```

Comment:
It would be good to keep the order of checks consistent with source table checks similar to `RenameMaterializedViewTask` so the first check for `isMaterializedView`, then `isView` and last `getTableHandle`

Reply:
Changed the order. Please have a look. Thanks!

Reply:
lgtm